documentation.md

```diff
@@ -0,0 +1,521 @@
+
+- _i_payload_ram_, responsible for storing the responses before they are transmitted to the requester. As there is one linked list per AXI identifier, the AXI identifier is not stored in the RAM, but deduced from its linked list identifier when it is released.
+- _i_meta_ram_out_tail_ and _i_meta_ram_out_head_, responsible for storing the linked list states: for each linked list element, they store the pointer to the next element in the payload RAM.
+
+Two RAM banks are used to hold pointer to next elements, as three ports are needed (which is explained by the linked list implementation below). Their content is therefore maintained identical, but they may be read at different addresses simultaneously.
+
+Using RAMs is efficient as it does not require a massive number of flip-flops to store data, but incurs one cycle latency for the output.
+
+<figure class="image">
+  <img src="https://i.imgur.com/mH3dPLo.png" alt="Simulated memory controller internal overview">
+  <figcaption>Fig: Response banks RAMs</figcaption>
+</figure>
+
+### Extended RAM cells
+
+Two options have been explored to store burst responses, while keeping a single internal identifier per address (and, therefore, per full burst of responses):
+
+1. _In-width burst storage_: Storing all the responses corresponding to the same burst in the same physical RAM cell, using masks to write and retrieve them.
+2. _In-depth burst storage_: Storing all the responses corresponding to the same burst in consecutive physical RAM cells.
+
+<figure class="image">
+  <img src="https://i.imgur.com/1yWktlc.png" alt="Simulated memory controller internal overview">
+  <figcaption>Fig: In-width burst storage (not chosen as implementation), for a burst length of 4</figcaption>
+</figure>
+
+The burst support has been implemented _in-depth_, because the block RAMs are typically narrow and deep, and do not always propose write masks, which makes in-width storage difficult.
+
+The _MaxBurstLen_ (maximum burst length for reads, and 1 for write responses) consecutive RAM cells dedicated to the same burst are referred to as _extended RAM cells_. On a high level, extended RAM cells are the elementary data structures in linked lists. BY opposition to extended RAM cells, the actual physical RAM cells are called _elementary RAM cells_.
+
+We additionally introduce two address spaces:
+
+- _address_: refers to the address of the extended cells, viewing them as elementary storage elements. This matches with addresses as defined by the metadata RAM.
+- _full address_: refers to the full address of an elementary cell in the payload RAM.
+
+<figure class="image">
+  <img src="https://i.imgur.com/VzySGSn.png" alt="Simulated memory controller internal overview">
+  <figcaption>Fig: In-depth burst storage, for a burst length of 4</figcaption>
+</figure>
+
+<figure class="image">
+  <img src="https://i.imgur.com/fzaWZ6v.png" alt="Simulated memory controller internal overview">
```

Comment:
Btw: can you add the SVGs for all figures under `/doc` in GitHub? This makes later modifications much easier. I think you can directly use SVG in MarkDown and don't need to go via PNG.